src/main/webapp/js/projects/react/components/ProjectsSearch.js

```diff
@@ -6,8 +6,15 @@ import ProjectList from './ProjectList.js';
 import {
   standardizeFetchErrors,
   makeRelativeUrlAbsolute,
-  basicErrorHandling,
 } from '../../../fetch_handler.js';
+import {DataFetcher} from '../../../common/react/components/DataFetcher.js';
+import ExpandedProject from './ExpandedProject.js';
+
+// Using HashRouter to avoid having to configure servlet routing:
+// https://stackoverflow.com/questions/27928372/react-router-urls-dont-work-when-refreshing-or-writing-manually
```

Comment:
Comments including links like this are helpful, though note previous guidance about comments explaining why/what the code is trying to accomplish, rather than trying to explain what the code is doing.

https://screenshot.googleplex.com/v62RVOLU2KZ

Not sure if this would fall into the category of "assume readers are familiar with React Router" which actually most of us working on this project aren't. But I'd consider putting helpful links like this in the PR rather than in the comments depending on how important you think it is for it to be right there with the code.

Reply:
That's a good point. I originally included the link because for those familiar with React Router they would have some immediate questions as to why I'm using the HashRouter here instead of the BrowserRouter, which is preferred. I'll take a pass over the whole file with your advice in mind, thanks for the help with this!